ast.ts

```diff
@@ -41,6 +43,7 @@ export type Expr<A> =
 export type Literal = 
     { tag: "num", value: number }
   | { tag: "bool", value: boolean }
+  | { tag: "string", value: string, length: number}
```

Comment:
Can we drop the `length` field of this `Literal` and just use `.length` when needed.

Reply:
Dropped length.